MaxText/train.py

```diff
@@ -305,13 +303,16 @@ def train_loop(config, state=None):
   if config.enable_profiler and first_profiling_step >= config.steps:
     raise ValueError("Profiling requested but initial profiling step set past training final step")
   last_profiling_step = np.clip(first_profiling_step + config.profiler_steps - 1, first_profiling_step, config.steps - 1)
+
   for step in np.arange(start_step, config.steps):
     if step == first_profiling_step:
       max_utils.activate_profiler(config)
 
     example_batch = load_next_batch(data_iterator, example_batch, config)
+
+    nextrng = jax.random.fold_in(init_rng, start_step)
```

Comment:
I think we want jax.random.fold_in(init_rng, step) so it generates new rng each step

Reply:
OMG thank you.